freud/order.pyx

```diff
@@ -494,6 +494,173 @@ cdef class TransOrderParameter:
+
+    The :data:`norm` constructor arguement provides
+    normalized versions of the plain :math:`Ql` or :math:`Wl` or normalized
+    average if the :data:`average` flag is set to true.  where the normalization
+    is performed by dividing by the average :math:`Q_{lm}` values over all
+    particles.
+
+    .. moduleauthor:: Xiyu Du <xiyudu@umich.edu>
+    .. moduleauthor:: Vyas Ramasubramani <vramasub@umich.edu>
+
+    Args:
+        box (:class:`freud.box.Box`):
+            Simulation box.
+        rmax (float):
+            Cutoff radius for the local order parameter. Values near the first
+            minimum of the RDF are recommended.
+        l (unsigned int):
+            Spherical harmonic quantum number l. Must be a positive number.
+        rmin (float):
+            Can look at only the second shell or some arbitrary RDF region.
+        average (bool):
+            Determines whether to calcuate the averaged Steinhardt order
+            parameter.
+        norm (bool):
+            Determines whether to calcuate the normalized Steinhardt order
+            parameter.
+        Wl (bool):
+            Determines whether to use the :math:`Wl` version of the Steinhardt
+            order parameter.
+            
+
+    Attributes:
+        box (:class:`freud.box.Box`):
+            Box used in the calculation.
+        num_particles (unsigned int):
+            Number of particles.
+        St (:math:`\left(N_{particles}\right)` :class:`numpy.ndarray`):
+            The last computed selected variant of the Steinhardt order
+            parameter for each particle (filled with NaN for particles with no
+            neighbors).
+    .. todo:: move box to compute, this is old API
+    """  # noqa: E501
+    cdef freud._order.Steinhardt * stptr
+    cdef freud.box.Box m_box
+    cdef rmax
+
+    def __cinit__(self, box, rmax, l, rmin=0, average=False, norm=False,
+            Wl=False, *args, **kwargs):
+        cdef freud.box.Box b = freud.common.convert_box(box)
+        if type(self) is Steinhardt:
+            self.m_box = b
+            self.rmax = rmax
+            self.stptr = new freud._order.Steinhardt(
+                dereference(b.thisptr), rmax, l, rmin,
+                average, norm, Wl)
+
+    def __dealloc__(self):
+        if type(self) is Steinhardt:
+            del self.stptr
+            self.stptr = NULL
+
+    @property
+    def box(self):
+        return freud.box.BoxFromCPP(<freud._box.Box> self.stptr.getBox())
+
+    @box.setter
+    def box(self, value):
+        cdef freud.box.Box b = freud.common.convert_box(value)
+        self.stptr.setBox(dereference(b.thisptr))
+
+    def setBox(self, box):
+        R"""Reset the simulation box.
+
+        Args:
+            box (:class:`freud.box.Box`): Simulation box.
+        """
+        self.box = box
+
+    @property
+    def num_particles(self):
+        cdef unsigned int np = self.stptr.getNP()
+        return np
+
+    @property
+    def St(self):
```

Comment:
I discussed with @vyasr a while back - I think we'd like to use the property name `order` for all order parameters going forward. Since this is a new class, let's rename `St` to `order`.